python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -517,23 +518,27 @@ def quantize(model: Model,
             return patch_attrs(result, model)
 
         elif precision == 'bf16':
-            invalidInputError(accelerator == 'openvino',
+            invalidInputError(accelerator == 'openvino' or accelerator is None,
                               "Accelerator {} is invalid for BF16.".format(accelerator))
             invalidInputError(device == 'CPU',
                               "Device {} don't support bfloat16.".format(device))
-            final_openvino_option = {"INFERENCE_PRECISION_HINT": "bf16"}
-            if openvino_config is not None:
-                final_openvino_option.update(openvino_config)
-            from bigdl.nano.deps.openvino.tf.model import KerasOpenVINOModel    # type: ignore
-            result = KerasOpenVINOModel(model,
-                                        input_spec=input_spec,
-                                        precision=precision,
-                                        thread_num=thread_num,
-                                        device=device,
-                                        config=final_openvino_option,
-                                        logging=logging,
-                                        **kwargs)
-            return patch_attrs(result, model)
+            if accelerator == 'openvino':
+                final_openvino_option = {"INFERENCE_PRECISION_HINT": "bf16"}
+                if openvino_config is not None:
+                    final_openvino_option.update(openvino_config)
+                from bigdl.nano.deps.openvino.tf.model import KerasOpenVINOModel    # type: ignore
+                result = KerasOpenVINOModel(model,
+                                            input_spec=input_spec,
+                                            precision=precision,
+                                            thread_num=thread_num,
+                                            device=device,
+                                            config=final_openvino_option,
+                                            logging=logging,
+                                            **kwargs)
+                return patch_attrs(result, model)
```

Comment:
should call `patch_compiled` before `patch_attrs`:
<https://github.com/intel-analytics/BigDL/blob/main/python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py#L648>